accepted/2020/mono-convergence/platform-specific-httpclient.md

```diff
@@ -0,0 +1,452 @@
+# Platform Specific HttpClient Support in dotnet/runtime
+
+## Overview
+The underlying networking libraries and capabilities in iOS, Android, and Wasm have required us to implement custom **HttpMessageHandler** classes using platform-specific APIs. Because each platform has unique API, different release pressures, and deeper infrastructure dependencies, the Mono team pushed the platform implementation into the individual SDKs. 
+
+The purpose of this document is to outline how such an approach could work for .NET 5 and help drive consensus.
+
+## Required Scenarios
+
+### Default HttpClient and HttpClientHandler will always use the most suitable HttpMessageHandler on the platform
+SocketHttpHandler, which is the current implementation under HttpClientHandler in dotnet/runtime, works great on desktop but it has limitations on platforms where sockets are not available or don’t offer the required level of control. The intention is to have the most common HttpClient construction to work out of the box on all platforms and use the platform most reliable HTTP provider.
+
+**Example**
+```c#
+// This should use the default message handler for the application. Either as
+// preselected by the framework or set in the Project Options for the project.
+
+HttpClient client = new HttpClient();
+
+// Same behavour should be observed when using following code
+HttpClient client = new HttpClient(new HttpClientHandler());
```

Comment:
I assume there are at least some properties on HttpClientHandler that won't work on all the platforms?  Might be worth calling that out here, that while `new HttpClient()` and `new HttpClient(new HttpClientHandler())` should work fine and identically, `new HttpClient(new HttpClientHandler() { SomeProperty = ... })` may not be supported everywhere, presumably throwing PlatformNotSupportedException where it's not.

Reply:
Included property sample